### PR TITLE
Feature request: Addresses issue #1137 Save window size on exit

### PR DESCRIPTION
**Pull request by willkillson**
Used electron-settings package to save the position and width/height of the application. The default width and height is 800 by 600 which was a value judgement. Please let me know if this will suffice, or if full screen is a better default.

closes https://github.com/azerothcore/Keira3/issues/1137

https://www.npmjs.com/package/electron-settings

Demo:
https://gyazo.com/aa4bfdefc9fa8fb1d3a59a615cf79c32

### CODE REVIEW

**Comment by Kitzunu** [comment]
Thanks for the PR!

A tip when you are fixing an open issue is to link it in your description as "Closes https://github.com/azerothcore/Keira3/issues/1137"

I have updated it for you this time =)

**Comment by 55Honey** [comment]
Definitely not fullscreen as a default imo. Imagine 43" of Keira 3 :-)

**Comment by willkillson** [comment]
> Definitely not fullscreen as a default imo. Imagine 43" of Keira 3 :-)

Thanks will do next time! Yeah I'm rocking the g9, I didn't like it either 😜

**Comment by Kitzunu** [comment]
When trying to build I get this error
```
main.ts:4:27 - error TS2307: Cannot find module 'electron-settings' or its corresponding type declarations.

4 import * as settings from 'electron-settings';
                            ~~~~~~~~~~~~~~~~~~~
```
Perhaps I am doing something wrong. Do you know?

**Comment by willkillson** [comment]
Normally typescript complains when you don't install the @types/<package>, but I didn't have that issue. I will bring in those types and see if that fixes this problem. Sorry about that 😬

**Review by FrancescoBorzi — APPROVED** [review]

File: main.ts [hunk 1]
```diff
@@ -48,6 +60,14 @@ function createWindow() {
   }
 
   win.on('close', function (e) {
+    //Save the settings
+    let bounds = win.getBounds();
```
Comment (FrancescoBorzi) [inline main.ts:64]: `bounds` can be `const`

File: main.ts [hunk 2]
```diff
@@ -10,14 +11,25 @@ const args = process.argv.slice(1);
...
+  let height = 600;
+  let pox_x = 0;
+  let pos_y = 0;
+
+  if (hasPreviousSettings) {
+    width = settings.getSync('user_settings.width') as number;
+    height = settings.getSync('user_settings.height') as number;
+    pox_x = settings.getSync('user_settings.pos_x') as number;
+    pos_y = settings.getSync('user_settings.pos_y') as number;
+  }
```
Comment (FrancescoBorzi) [inline main.ts:25]:
```suggestion
  const width = hasPreviousSettings ? Number(settings.getSync('user_settings.width')) : 800;
  const height = hasPreviousSettings ? Number(settings.getSync('user_settings.height')) : 600;
  const pox_x = hasPreviousSettings ? Number(settings.getSync('user_settings.pos_x')) : 0;
  const pos_y = hasPreviousSettings ? Number(settings.getSync('user_settings.pos_y')) : 0;
```

**Comment by willkillson** [comment]
@FrancescoBorzi yes! Those turnery's are much better. 
Changed that line to const.

**Comment by FrancescoBorzi** [comment]
thank you @willkillson !

**Pull request merged**